sklearn_numba_dpex/common/tests/test_random.py

```diff
@@ -74,7 +82,65 @@ def test_xoroshiro128pp():
         n_states=1, seed=42, subsequence_start=1
     )
     actual_res_2 = [
-        int(dpt.asnumpy(get_random_raw(random_state))[0]) for i in range(10)
+        int(dpt.asnumpy(get_random_raw(random_state))[0]) for _ in expected_res_2
     ]
 
     assert expected_res_2 == actual_res_2
+
+
+def _make_get_single_rand_value_kernel(dtype):
+
+    rand_uniform_kernel_func = make_rand_uniform_kernel_func(np.dtype(dtype))
+    zero_idx = np.int64(0)
+
+    @dpex.kernel
+    # fmt: off
+    def _get_single_rand_value(
+        random_state,                     # IN             (1, 2)
+        single_rand_value,                # OUT            (1,)
+        ):
+        single_rand_value[0] = rand_uniform_kernel_func(random_state, zero_idx)
+
+    def get_single_rand_value(random_state):
+        single_rand_value = dpt.empty(sh=1, dtype=dtype)
+        _get_single_rand_value[1, 1](random_state, single_rand_value)
+        return dpt.asnumpy(single_rand_value)[0]
+
+    return get_single_rand_value
+
+
+def test_xoroshiro128pp_rand():
+    random_state = create_xoroshiro128pp_states(n_states=1, seed=42)
+    get_single_rand_value_kernel_32 = _make_get_single_rand_value_kernel(np.float32)
+
+    expected_res_32 = [
+        0.9083704,
+        0.33061236,
+        0.9509354,
+        0.20447117,
+        0.34282982,
+        0.09195548,
+        0.49001336,
+        0.09946841,
+        0.23281485,
+        0.3062514,
+    ]
+    expected_res_32 = [np.float32(f) for f in expected_res_32]
+
+    actual_res_32 = [
+        get_single_rand_value_kernel_32(random_state) for _ in expected_res_32
+    ]
+
+    assert expected_res_32 == actual_res_32
+
+    if not random_state.device.sycl_device.has_aspect_fp64:
+        return
+
+    random_state = create_xoroshiro128pp_states(n_states=1, seed=42)
+    get_single_rand_value_kernel_64 = _make_get_single_rand_value_kernel(np.float64)
+    actual_res_64_to_32 = [
+        np.float32(get_single_rand_value_kernel_64(random_state))
+        for _ in expected_res_32
+    ]
+
+    np.testing.assert_allclose(actual_res_64_to_32, expected_res_32, rtol=1e-4)
```

Comment:
Should we preferably use fixtures from scikit-learn for tests assertions (i.e. [`sklearn.utils._testing.assert*`](https://github.com/scikit-learn/scikit-learn/blob/main/sklearn/utils/_testing.py))?

Reply:
I didn't want to add a dependency on sklearn but that's dumb sklearn is already a mandatory dependency :facepalm: will fix & merge

Reply:
Do not :facepalm: yourself: it's fine. :slightly_smiling_face:

Reply:
I am reviewing this now. Please wait for my review before merging.